packages/hop-node/src/watchers/BondWithdrawalWatcher.ts

```diff
@@ -371,8 +371,14 @@ class BondWithdrawalWatcher extends BaseWatcher {
       await destinationBridge.provider.call(tx)
       return true
     } catch (err) {
-      logger.error(`getIsRecipientReceivable err: ${err.message}`)
-      return false
+      const revertErrMsgRegex = /(execution reverted)/i
```

Comment:
It is expected that the string `execution reverted` would never get returned in the case of a server error correct?

For example, when the rpc endpoint is offline then the ethers error contains `the execution failed due to an exception.","data":"Reverted"}` which can be misleading because it never hit the contract (it doesn't contain the exact string `execution reverted` but it's similar)

```bash
processing response error (body="{"jsonrpc":"2.0","error":{"code":-32016,"message":"The execution failed due to an exception.","data":"Reverted"},"id":695}n", error= { "code": -32016, "data": "Reverted" } , requestBody="{"method":"eth_estimateGas","params":[{"gasPrice":"0x430e23400","from":"0xa6a688f107851131f0e1dce493ebbebfaf99203e","to":"0x25d8039bb044dc227f741a9e381ca4ceae2e6ae8","data":"0x3d12a85a0000000000000000000000007ee5515dd8ca27afad5277ea07d5065034ed6df000000000000000000000000000000000000000000000000000000000001eabde61702a46938da1d8e1ad6aa98df8a7ef77b2bcc5dab6981b5a98798635b7fcfe00000000000000000000000000000000000000000000000000000000000f42bb0000000000000000000000000000000000000000000000000000000000000bc40000000000000000000000000000000000000000000000000000000061730a43"}],"id":695,"jsonrpc":"2.0"}", requestMethod="POST", url="https://xdai.rpc.hop.exchange", code=SERVER_ERROR, version=web/5.4.0)
```

It'd be good to know if the error string is consistent across providers and won't return in the case of an actual server error (I don't think it will though)

Reply:
That was the intention. It looks like Alchemy/Infura provide that error, but xDai does not.

Alchemy: `execution reverted`
```
UnhandledPromiseRejectionWarning: Error: missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\": \"2.0\", \"id\": 44, \"error\": {\"code\": -32000, \"message\": \"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"value\":\"0x1\",\"from\":\"0x0000000000000000000000000000000000000000\",\"to\":\"0xb6e90ebe44de40aeb0b987adc2d7d9dd0ec918d7\"},\"latest\"],\"id\":44,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"<RPC_ENDPOINT>"}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.0)
```

Infura: `execution reverted`
```
UnhandledPromiseRejectionWarning: Error: missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":44,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"value\":\"0x1\",\"from\":\"0x0000000000000000000000000000000000000000\",\"to\":\"0xb6e90ebe44de40aeb0b987adc2d7d9dd0ec918d7\"},\"latest\"],\"id\":44,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"<RPC_ENDPOINT>"}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.0)
```

Quicknode: `VM execution error`
```
UnhandledPromiseRejectionWarning: Error: missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32015,\"message\":\"VM execution error.\",\"data\":\"Reverted 0x\"},\"id\":44}\n","error":{"code":-32015,"data":"Reverted 0x"},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"value\":\"0x1\",\"from\":\"0x0000000000000000000000000000000000000000\",\"to\":\"0x415bc92d42dbc7397fb1a1c0ddb40780a9d58ed1\"},\"latest\"],\"id\":44,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"<RPC_ENDPOINT>"}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.0)
```

Resolved with a5135ee8c9aa096c2a57a30a5796f0f002ff194b